### ISSUE TITLE
[jsp] Invalid JSP/HTML file is not detected

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1261]
**Affects PMD Version:** 6.6.0

**Rule:** n/a

**Description:**
If the closing tags don't appear in order, the HTML file can be considered invalid and we should probably throw a parsing exception.

There is a exception class `StartAndEndTagMismatchException`, which is not used, but fits this use case...

Question:
* Should this be always a error? Or should we be "friendly"? Maybe this should be an option (strict parsing vs. loose parsing)?

**Code Sample demonstrating the issue:**

```
<html>
  <head/>
  <body>
    <div><p>
  </body>
  </div> <!-- this is misplaced... should be before the closed body -->
</html>
```

**Running PMD through:** *Designer*
